src/finn/transformation/lower_convs_to_matmul.py

```diff
@@ -243,5 +240,4 @@ def apply(self, model):
                 # remove old nodes
                 graph.node.remove(n)
 
-        model = model.transform(InferShapes())
```

Comment:
Could you explain why the call to `InferShapes()` is removed here?

Reply:
If I am not mistaken, when the convolution is lowered, additional nodes and value infos are inserted. These new value infos are created with the 'make_tensor_value_info' function from onnx/helper.py (https://github.com/onnx/onnx/blob/b52f029f4b76904b148d4defde8c43bc534feb59/onnx/helper.py#L431). All of these value infos are constructed by also passing a shape variable. The initializer of the MatMul node is set with the 'set_initializer' method from finn/core/modelwrapper.py. A value info is then created by a call to the method 'set_tensor_shape', which in turn calls 'make_tensor_value_info' (https://github.com/Xilinx/finn-base/blob/aea01d6c4f02e3c1d699af6b367faefccd6cf4f8/src/finn/core/modelwrapper.py#L272). 

Therefore, it seemed like the InferShapes transformation will not add anything to the existing value infos, as all of them will be created with a shape annotation.
However, the InferShapes transformation could be useful to catch an unexpected convolution configuration, as the shape of the output of the Im2Col node will not match the shape of the Conv node output, which will result in an error. Ideally, such unexpected configurations should be detected in the lowering transformation (and trigger an error). However, looking back to it, it could be useful to keep the InferShapes transformation just to catch any unexpected (edge case) convolution configurations that were not spotted by the lowering transformation.

There is a certain trade-off that I had to make: either have the transformation to catch unexpected (edge case) convolutions that cannot be lowered at the cost of (a slight) increase in execution time, or leave the transformation out and rely on the lowering transformation to catch all unexpected (edge case) convolutions.

This is how I looked at the InferShapes transformation. What do you recommend?

Reply:
If we can guarantee that all new intermediate tensors created by the transform already have shapes this is fine. There are also plans to apply "tid-up" transformations like shape inference after every other transformation automatically, so this is also cleaner from that perspective. Let's remove it for now and see if any of the existing FINN examples error out because of this once the `finn-base` commit hash is updated.